File: ClientApp/src/app/shared/models/wishlist.ts

export interface Wishlist {
  id: string;
  name: string;
  userId: string;
  username: string;
  wishListItems: WishlistItem[];
}

export interface WishlistItem {
  id: string;
  clothingItemName: string;
  clothingItemId: string;
  pictureUrl: string;
}

File: src/FashionClothesAndTrends.Domain/Entities/Wishlist.cs
using FashionClothesAndTrends.Domain.Common;

namespace FashionClothesAndTrends.Domain.Entities;

public class Wishlist : BaseEntity
{
    public required string UserId { get; set; }
    public User User { get; set; } = null!;
    public required string Name { get; set; }
    public ICollection<WishlistItem> Items { get; set; } = [];
}